### ISSUE TITLE
Option to truncate excessively large `.html` in results

### ISSUE THREAD

**Issue by stephenmathieson** [dequelabs/axe-core#4544]
### Product

axe-core

### Feature Description

When analyzing a axe Watcher results in prod, we noticed we have _a few_ excessively large datums (>1MB). This is preventing us from migrating result data to a different (_much faster_ data store).

After digging further, it was noticed that this is due to axe-core including the element in violation's `outerHTML` (eg `.violations[x].nodes[y].html`) and its related node's `outerHTML` (`.violations[x].nodes[y].relatedNodes[z].html`). Consider the following snippet:

```html
<img class="foo" src="data:png,base64, EXTREMELY LARGE IMAGE CONVERTED TO BASE64" />
```

It appears [axe-core is already truncating _some_ parts (children) of the element's HTML](https://github.com/dequelabs/axe-core/blob/v4.9.1/lib/core/utils/dq-element.js#L10-L19), but not the element itself.

An option similar to `axe.configure({ noHtml: true })` which causes axe-core to truncate *ALL* HTML snippets would be extremely helpful.

**Comment by WilcoFiers** [comment]
@stephenmathieson Thanks for the suggestion. That's definitely a problem we should find a solution to. I'm not sure whether it should be a separate option. Axe-core truncates HTML snippets to prevent this problem. I think what we might want to do also put limits on attributes, both on the length of an attribute value, and of the number of attributes.

I'll bring this up with the team to come up with a solution.

**Comment by WilcoFiers** [comment]
Talked about this, we agree this should happen. Will be in 4.11 / 5.0, won't do this in a patch.

We think the best way to do this is to:
1. Look at each attribute, if any are too long replace the value with "..."
2. If this results in a string that is overall still too long, start removing attributes and ad "..." at the end of the tag.

**Comment by namansancheti** [comment]
Hi @WilcoFiers,
I'm new to open-source contributions and this would be my first contribution to `axe-core`, but I would like to work on this - can you please confirm no one else has already started on this ?

**Comment by padmavemulapati** [comment]
This feature is released already and the testable html content file is not in QA-scope (didnot find a visible change in any axe-core supported rules)
